### PR TITLE
[v2] Update Project Route

### PR DESCRIPTION
**Pull request by harshkhandeparkar**
#### Changes
- Added the `/project/update` route.

### CODE REVIEW

File: controllers/project_update.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	project := models.Project{}
+	tx := db.
+		Table("projects").
+		Where("id = ?", reqFields.Id).
+		First(&project)
+
+	if tx.Error != nil && tx.Error != gorm.ErrRecordNotFound {
+		utils.LogErrAndRespond(r, w, err, "Database error.", http.StatusInternalServerError)
+		return
+	} else if tx.Error == gorm.ErrRecordNotFound {
```
Comment (rajivharlalka) [inline controllers/project_update.go:74]:
```suggestion
	} 
	if tx.Error == gorm.ErrRecordNotFound {
```

File: controllers/project_update.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+		utils.LogWarnAndRespond(
```
Comment (rajivharlalka) [inline controllers/project_update.go:75]: Shouldn't this be an error too?

File: controllers/project_update.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+	updatedProj := &models.Project{
+		Name:               reqFields.Name,
+		Desc:               reqFields.Description,
+		Tags:               reqFields.Tags,
+		RepoLink:           reqFields.RepoLink,
+		ComChannel:         reqFields.ComChannel,
+		README:             reqFields.ReadmeURL,
+		SecondaryMentor_id: int32(secondaryMentor.ID),
+	}
```
Comment (rajivharlalka) [inline controllers/project_update.go:117]: is 0 the default secondary mentor id?

File: controllers/project_update.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+			r.RequestURI,
+			"POSSIBLE SESSION HIJACKING.",
+			fmt.Sprintf("JWT Username: %s", login_username),
+			fmt.Sprintf("Given Username: %s", reqFields.MentorUsername),
+		)
+
+		w.WriteHeader(http.StatusUnauthorized)
+		fmt.Fprint(w, "Login username and mentor username do not match.")
+		return
+	}
```
Comment (rajivharlalka) [inline controllers/project_update.go:62]: This seems quite redundant across multiple routes, can add in the middleware itself

**Review by rajivharlalka — CHANGES_REQUESTED** [review]

File: controllers/project_update.go (hunk 4 above)
Comment (harshkhandeparkar) [reply controllers/project_update.go:62]: #130

Comment (harshkhandeparkar) [reply controllers/project_update.go:62]: Tracked here.

File: controllers/project_update.go (hunk 3 above)
Comment (harshkhandeparkar) [reply controllers/project_update.go:117]: I think yes

File: controllers/project_update.go (hunk 2 above)
Comment (harshkhandeparkar) [reply controllers/project_update.go:76]: Changing

Comment (harshkhandeparkar) [reply controllers/project_update.go:76]: Done.

File: controllers/project_update.go (hunk 3 above)
Comment (rajivharlalka) [reply controllers/project_update.go:120]: confirm once, else need a check here otherwise wrong mentor would be asisgned

Comment (harshkhandeparkar) [reply controllers/project_update.go:120]: Confirmed. The default is 0 and the IDing starts at 1 so a wrong mentor would never be assigned.

**Review by rajivharlalka — APPROVED** [review]

**Pull request merged**